### PR TITLE
DYN-7400 : remove python runtime dependency

### PR DESCRIPTION
**Pull request by BogdanZavu**
### Purpose
We want to remove Python.Runtime dependency at D4R level in order to easily handle multiple python engines.
The dictionary marshaler has been moved at DynamoCore level - will add reference to the new DynamoServices package that contains the coresponding API once [this pr](https://github.com/DynamoDS/Dynamo/pull/15537) is merged and a new package is published.

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] Snapshot of UI changes, if any.

### Reviewers
@Mikhinja @twastvedt

### CODE REVIEW

**Review by twastvedt — APPROVED** [review]
Makes sense to me, but I'm no marshaling expert.

**Review by Mikhinja — APPROVED** [review]

File: src/DynamoRevit/Models/RevitDynamoModel.cs [hunk 1]
```diff
@@ -495,56 +505,16 @@ private void OnPythonEngineCollectionChanged(object sender, NotifyCollectionChan
...
-                        });
-            }
-            Func<object, object> unwrap = UnwrapElementMarshaler.Marshal;
             // Turn off element binding during python script execution
             ElementBinder.IsEnabled = false;
-            // register UnwrapElement method
-            scopeSet("UnwrapElement", unwrap);
+            if (UnwrapElementMarshaler != null)
+            {
+                Func<object, object> unwrap = UnwrapElementMarshaler.Marshal;
```
Comment (Mikhinja) [inline src/DynamoRevit/Models/RevitDynamoModel.cs:514]: Do nested `dict` and `list` objects work appropriately?

Do we know if there were existing workarounds related to python wrapping done by 3rd party packages (nodes) or included in python scripts?

Comment (BogdanZavu) [reply src/DynamoRevit/Models/RevitDynamoModel.cs:514]: I think yes. I've been looking into some graphs from the tests in D4R that make use of that code and it seems to work as expected.

**Comment by Mikhinja** [comment]
If this requires Dynamo 3.4 then I recommend waiting for Dynamo 3.4 to be integrated first before merging

File: src/Config/packages.aget [hunk 2]
```diff
@@ -2,13 +2,13 @@
...
     "references": {
       "DynamicLanguageRuntime": "1.2.2",
-      "DynamoVisualProgramming.Core": "3.3.0.6316",
-      "DynamoVisualProgramming.DynamoCoreNodes": "3.3.0.6316",
-      "DynamoVisualProgramming.DynamoServices": "3.3.0.6316",
-      "DynamoVisualProgramming.Tests": "3.3.0.6316",
-      "DynamoVisualProgramming.WpfUILibrary": "3.3.0.6316",
-      "DynamoVisualProgramming.ZeroTouchLibrary": "3.3.0.6316",
-      "DynamoVisualProgramming.DynamoSamples": "3.3.0.6316",
+      "DynamoVisualProgramming.Core": "3.4.0-beta6616",
```
Comment (Mikhinja) [inline src/Config/packages.aget:5]: In order to merge in Revit we need to use a build that has retention "keep" in artifactory. This build currently has retention policy 30 ( days ).

Comment (BogdanZavu) [reply src/Config/packages.aget:5]: Good catch! I changed the policy to keep.

**Pull request merged**